popup.js

```diff
@@ -118,16 +111,18 @@ function savesnaps(){
 
 // Add date and time to text input
 function adddate(){
-
     var months=['Jan','Feb','Mar','Apr','May','Jun','Jul','Aug','Sep','Oct','Nov','Dec'];
     var dat=new Date();
     var d=dat.getDate();
     var m=months[dat.getMonth()];
     var hr=dat.getHours();
-    if (hr>12){hr=hr-12+":"+dat.getMinutes()+" PM";}
-    else if (hr==12){hr=hr+":"+dat.getMinutes()+" PM";}
-    else {hr=hr+":"+dat.getMinutes()+" AM";}
-    console.log(d);
+    var min=dat.getMinutes();
+    if(d<10){d="0"+d;}
+    if(min<10){min="0"+min;}
+    if (hr>12){hr=hr-12+":"+min+" PM";}
+    else if (hr==12){hr=hr+":"+min+" PM";}
+    else {hr=hr+":"+min+" AM";}
+    if (parseInt(hr)<12){hr= "0"+hr}
```

Comment:
It should be less than 10, otherwise for `10:15` or `11:32`, it will use `010:15` and `011:32` respectively. Also, why are you parsing a `string` as `int`? Technically, it works, but it's not much logical.
Try to make required changes with `hr` when it's `int` and then add the `min` and `AM/PM` as `string` to it.

Reply:
I'm talking about line 125. It should be < 10 there.